File: backend/transaction-service/transaction-domain/transaction-domain-core/src/main/java/com/saver/system/transaction/service/domain/event/TransactionEvent.java
package com.saver.system.transaction.service.domain.event;

import com.saver.system.domain.event.DomainEvent;
import com.saver.system.transaction.service.domain.entity.Transaction;

public abstract class TransactionEvent implements DomainEvent<Transaction> {
    private final Transaction transaction;
    public TransactionEvent(Transaction transaction) {
        this.transaction = transaction;

    }
    public Transaction getTransaction() {
        return transaction;
    }
}

File: backend/Services/transactions/src/domain/domain-core/event/transaction-event.ts
import { DomainEvent } from "@machsan-tiggo/common";
import { Transaction } from "../entity/transaction";

export abstract class TransactionEvent implements DomainEvent<Transaction> {
  protected transaction: Transaction;

  public constructor(transaction: Transaction) {
    this.transaction = transaction;
  }

  public getTransaction(): Transaction {
    return this.transaction;
  }
}
